src/stream/src/executor/join/hash_join.rs

```diff
@@ -1196,6 +978,182 @@ where {
     }
 }
 
+pub struct AsOfJoinHashMap<S: StateStore> {
```

Comment:
We should decouple this. We can have a module for `as_of_join`, independent of `hash_join`.